src/process/receipts.rs

```diff
@@ -0,0 +1,297 @@
+                    tx_hashes_for_receipts
+                        .get(r.receipt_id.to_string().as_str())
+                        .expect("transaction hash expected as we have check for it earlier")
+                        .clone(),
+                ))
+            } else {
+                warn!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Skipping Receipt {} as we can't find parent Transaction for it.",
+                    r.receipt_id.to_string()
+                );
+                skipping_receipt_ids.push(r.receipt_id.clone());
+                None
+            }
+        })
+        .collect();
+
+    let save_receipts_future = save_receipts(&pool, receipt_models);
+
+    let receipts_to_process: Vec<&near_indexer::near_primitives::views::ReceiptView> = receipts
+        .into_iter()
+        .filter(|r| !skipping_receipt_ids.contains(&r.receipt_id))
+        .collect();
+
+    let process_receipt_actions_future = process_receipt_actions(&pool, &receipts_to_process);
+
+    let process_receipt_data_future = process_receipt_data(&pool, &receipts_to_process);
+
+    join!(
+        save_receipts_future,
+        process_receipt_actions_future,
+        process_receipt_data_future
+    );
+}
+
+async fn save_receipts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    receipts: Vec<models::Receipt>,
+) {
+    loop {
+        match diesel::insert_into(schema::receipts::table)
+            .values(receipts.clone())
+            .execute_async(&pool)
+            .await
+        {
+            Ok(_) => break,
+            Err(async_error) => {
+                error!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Error occurred while Receipt were adding to database. Retrying in {} milliseconds... \n {:#?} \n{:#?}",
+                    crate::INTERVAL.as_millis(),
+                    async_error,
+                    receipts,
+                );
+                tokio::time::delay_for(crate::INTERVAL).await;
+            }
+        };
+    }
+}
+
+async fn process_receipt_actions(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    receipts: &[&near_indexer::near_primitives::views::ReceiptView],
+) {
+    let mut receipt_actions: Vec<models::ReceiptAction> = vec![];
+    let mut receipt_action_actions: Vec<models::ReceiptActionAction> = vec![];
+    let mut receipt_action_input_data: Vec<models::ReceiptActionInputData> = vec![];
+    let mut receipt_action_output_data: Vec<models::ReceiptActionOutputData> = vec![];
+
+    for receipt in receipts {
+        if let near_indexer::near_primitives::views::ReceiptEnumView::Action {
+            actions,
+            input_data_ids,
+            output_data_receivers,
+            ..
+        } = &receipt.receipt
+        {
+            match models::ReceiptAction::try_from(*receipt) {
+                Ok(model) => receipt_actions.push(model),
+                Err(_) => continue,
+            }
```

Comment:
Can we refactor it into:

```suggestion
        if let Ok(model) = models::ReceiptAction::try_from(*receipt) {
            receipt_actions.push(model),
```

?